### ISSUE TITLE
is_* wrong_self_convention false positive.

### ISSUE THREAD

**Issue by Aaronepower** [Manishearth/rust-clippy#1420]
This code triggers `wrong_self_convention` on `is_json` when this code is using is_* correctly.
```rust
use self::Format::*;

#[derive(Clone, Copy, Debug, PartialEq, Eq)]
pub enum Format {
    Json,
    Xml,
}

impl Format {
    pub fn is_json(&self) -> bool {
        match *self {
            Json => true,
            _ => false,
        }
    }
}

impl Default for Format {
    fn default() -> Self {
        Json
    }
}
```
